pkg/resourcemanager/azuresql/azuresqlmanageduser/azuresqlmanageduser_reconcile.go

```diff
@@ -146,7 +146,24 @@ func (s *AzureSqlManagedUserManager) Delete(ctx context.Context, obj runtime.Obj
 		requestedUsername = instance.Name
 	}
 
-	db, err := s.ConnectToSqlDbAsCurrentUser(ctx, DriverName, instance.Spec.Server, instance.Spec.DbName)
+	// short circuit connection if database doesn't exist
+	_, err = s.GetDB(ctx, instance.Spec.ResourceGroup, instance.Spec.Server, instance.Spec.DbName)
+	if err != nil {
+		instance.Status.Message = err.Error()
+
+		catch := []string{
+			errhelp.ResourceNotFound,
+			errhelp.ParentNotFoundErrorCode,
+			errhelp.ResourceGroupNotFoundErrorCode,
+		}
+		azerr := errhelp.NewAzureErrorAzureError(err)
+		if helpers.ContainsString(catch, azerr.Type) {
+			return false, nil
+		}
+		return false, err
```

Comment:
Sorry I missed this first time around, but since this case is a delete success (because the DB's gone) I think you need to do the secret deletion for it as well.

Reply:
Woaah, I totally missed that. Good catch.